packages/react/src/components/textInput/TextInput.stories.tsx

```diff
@@ -115,3 +117,53 @@ Playground.args = {
     'Tooltips contain "nice to have" information. Default Tooltip contents should not be longer than two to three sentences. For longer descriptions, provide a link to a separate page.',
   tooltipButtonAriaLabelText: 'Tooltip',
 };
+
+export const SimpleSearchInput = (args) => {
+  const [inputValue, setInputValue] = React.useState('');
+  const ref = useRef(null);
+
+  const doSearch = (value) => {
+    action('search for')(value);
+  };
+
+  const onButtonClick = () => {
+    doSearch(ref?.current?.value);
+  };
+
+  const onChangeHandler = (e) => {
+    setInputValue(e.target.value);
+  };
+
+  const onClearButtonClick = () => {
+    action('clear search')();
+    const nativeInputValueSetter = Object?.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, 'value').set;
+    nativeInputValueSetter?.call(ref.current, '');
+    ref?.current.focus();
+    ref?.current.dispatchEvent(new Event('change', { bubbles: true }));
+  };
```

Comment:
In my other comment I mentioned that this callback should be discarded.

If still kept, then this could be simplified with:

```
  const onClearButtonClick = () => {
    action('clear search')();
    const input = ref?.current as HTMLInputElement;
    setInputValue('');
    input.focus();
  };
```

 This requires the textInput to be controlled, but that should not be a problem
 
 `value={inputValue}`